recipes/libaom-av1/all/conanfile.py

```diff
@@ -32,16 +36,29 @@ def _source_subfolder(self):
     def _build_subfolder(self):
         return "build_subfolder"
 
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
     def config_options(self):
-        if self.settings.os == 'Windows':
+        if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
+        if self.settings.arch not in ("x86", "x86_64"):
+            del self.options.assembly
+
+    def build_requirements(self):
+        if self.options.get_safe("assembly", False):
+            self.build_requires("nasm/2.15.05")
+        if self._settings_build.os == "Windows":
+            self.build_requires("strawberryperl/5.30.0.1")
+
+    def validate(self):
```

Comment:
I would have kept validate() before build_requirements().

Reply:
When is `validate` called?
Can `validate` check for options of build requirements?